ci-operator/templates/openshift/openshift-azure/build-base-image.yaml

```diff
@@ -0,0 +1,194 @@
+# This template is being used for testing OpenShift Ansible base image builds for Azure.
+# At the moment it cannot be reused to launch cluster for Origin tests
+kind: Template
+apiVersion: template.openshift.io/v1
+
+parameters:
+- name: JOB_NAME_SAFE
+  required: true
+- name: JOB_NAME_HASH
+  required: true
+- name: NAMESPACE
+  required: true
+- name: CLUSTER_TYPE
+  value: "azure"
+  required: true
+- name: AZURE_VM_SIZE
+  value: "Standard_D2s_v3"
+  required: true
+- name: IMAGE_ANSIBLE
+  required: true
+- name: DEPLOY_OS
+  value: "rhel7"
+  required: true
+
+objects:
+
+# We want the cluster to be able to access these images
+- kind: RoleBinding
+  apiVersion: authorization.openshift.io/v1
+  metadata:
+    name: ${JOB_NAME_SAFE}-image-puller
+    namespace: ${NAMESPACE}
+  roleRef:
+    name: system:image-puller
+  subjects:
+  - kind: SystemGroup
+    name: system:unauthenticated
+
+# The build pod will create an instance and install openshift
+- kind: Pod
+  apiVersion: v1
+  metadata:
+    name: ${JOB_NAME_SAFE}
+    namespace: ${NAMESPACE}
+    annotations:
+      # we want to gather the teardown logs no matter what
+      ci-operator.openshift.io/wait-for-container-artifacts: teardown
+  spec:
+    restartPolicy: Never
+    activeDeadlineSeconds: 10800
+    terminationGracePeriodSeconds: 600
+    volumes:
+    - name: artifacts
+      emptyDir: {}
+    - name: shared-tmp
+      emptyDir: {}
+    - name: shared-data
+      emptyDir: {}
+    - name: cluster-secrets-azure
+      secret:
+        secretName: e2e-azure-secret
+    - name: openshift-azure
+      emptyDir: {}
+
+    containers:
+    - name: buildimage
+      image: ${IMAGE_ANSIBLE}
+      resources:
+        requests:
+          cpu: 1
+          memory: 300Mi
+        limits:
+          cpu: 3
+          memory: 4Gi
+      volumeMounts:
+      - name: shared-tmp
+        mountPath: /tmp/shared
+      - name: cluster-secrets-azure
+        mountPath: /usr/share/ansible/openshift-ansible/inventory/dynamic/injected
+      env:
+      - name: HOME
+        value: /usr/share/ansible/openshift-ansible
+      - name: AZURE_VM_SIZE
+        value: "${AZURE_VM_SIZE}"
+      - name: JOB_NAME_SAFE
+        value: "${JOB_NAME_SAFE}"
+      - name: JOB_NAME_HASH
+        value: "${JOB_NAME_HASH}"
+      - name: DEPLOY_OS
+        value: "${DEPLOY_OS}"
+      - name: TYPE
+        value: "${CLUSTER_TYPE}"
+      - name: INSTANCE_PREFIX
+        value: ${NAMESPACE}-${JOB_NAME_HASH}
+      command:
+      - /usr/local/bin/entrypoint-provider
+      args:
+      - /bin/bash
+      - -c
+      - |
+        #!/bin/bash
+        # error handling and sync code
+        set -euo pipefail
+
+        # We need to persist this variable so that we can clean it up when we exit
+        echo "RESOURCE_GROUP=${INSTANCE_PREFIX}-$(TZ=Etc/UTC date +%Y%m%d%H%M)" >>/tmp/shared/environment
```

Comment:
You don't need to share these two envs across containers now - can remove.

Reply:
I think this is still necessary. Since they are composed of time components (up to the minute) we cannot recompute those values in the teardown container. The teardown container will end up trying to delete a non-existent resource group.

Reply:
You are right about the resource group. You probably don't need IMAGE_NAME?

Reply:
sharing `IMAGE_NAME` is indeed no longer necessary